crates/cast/bin/cmd/creation_code.rs

```diff
@@ -0,0 +1,159 @@
+use alloy_primitives::{Address, Bytes};
+use alloy_provider::{ext::TraceApi, Provider};
+use alloy_rpc_types::trace::parity::{Action, CreateAction, CreateOutput, TraceOutput};
+use cast::SimpleCast;
+use clap::{command, Parser};
+use eyre::{eyre, OptionExt, Result};
+use foundry_block_explorers::Client;
+use foundry_cli::{
+    opts::{EtherscanOpts, RpcOpts},
+    utils,
+};
+use foundry_common::provider::RetryProvider;
+use foundry_config::Config;
+
+use super::interface::fetch_abi_from_etherscan;
+
+/// CLI arguments for `cast creation-code`.
+#[derive(Parser)]
+pub struct CreationCodeArgs {
+    /// An Ethereum address, for which the bytecode will be fetched.
+    contract: Address,
+
+    /// Disassemble bytecodes into individual opcodes.
+    #[arg(long)]
+    disassemble: bool,
+
+    /// Return creation bytecode without constructor arguments appended.
+    #[arg(long)]
+    without_args: bool,
+
+    /// Return only constructor arguments.
+    #[arg(long)]
+    only_args: bool,
+
+    #[command(flatten)]
+    etherscan: EtherscanOpts,
+    #[command(flatten)]
+    rpc: RpcOpts,
+}
+
+impl CreationCodeArgs {
+    pub async fn run(self) -> Result<()> {
+        let Self { contract, etherscan, rpc, disassemble, without_args, only_args } = self;
+
+        if without_args && only_args {
+            return Err(eyre!("--without-args and --only-args are mutually exclusive."));
+        }
```

Comment:
Use the `conflicts_with = "foobarbaz"` argument in `#[arg(long)]` of either to indicate that the two flags are conflicting. Note you only have to do this for one of the args, the other is inferred.